plugins/python-build/scripts/add_miniconda.py

```diff
@@ -0,0 +1,265 @@
+            return PyVersion.PY36
+        else:
+            return PyVersion.PY37
+
+    def with_version_triple(self):
+        return MinicondaVersion(
+            self.suffix, VersionStr.from_info(self.version_str.info()[:3])
+        )
+
+
+class MinicondaSpec(NamedTuple):
+    version: MinicondaVersion
+    os: SupportedOS
+    arch: SupportedArch
+    md5: str
+    py_version: Optional[PyVersion] = None
+
+    @classmethod
+    def from_filestem(cls, stem, md5, py_version=None):
+        miniconda_n, ver, os, arch = stem.split("-")
+        spec = MinicondaSpec(
+            MinicondaVersion(Suffix(miniconda_n[-1]), VersionStr(ver)),
+            SupportedOS(os),
+            SupportedArch(arch),
+            md5,
+        )
+        if py_version is None:
+            spec = spec.with_py_version(spec.version.default_py_version())
+        return spec
+
+    def to_install_lines(self):
+        return install_line_fmt.format(
+            repo=MINICONDA_REPO,
+            suffix=self.version.suffix,
+            version_str=self.version.version_str,
+            os=self.os,
+            arch=self.arch,
+            md5=self.md5,
+            py_version=self.py_version,
+        )
+
+    def with_py_version(self, py_version: PyVersion):
+        return MinicondaSpec(*self[:-1], py_version=py_version)
+
+    def with_version_triple(self):
+        version, *others = self
+        return MinicondaSpec(version.with_version_triple(), *others)
+
+
+def make_script(specs: List[MinicondaSpec]):
+    install_lines = [s.to_install_lines() for s in specs]
+    return install_script_fmt.format(install_lines="\n".join(install_lines))
+
+
+def get_existing_minicondas():
+    for p in out_dir.iterdir():
+        name = p.name
+        if not p.is_file() or not name.startswith("miniconda"):
+            continue
+        try:
+            v = MinicondaVersion.from_str(name)
+            if v.version_str != "latest":
+                yield v
+        except ValueError:
+            pass
+
+
+def get_available_minicondas():
+    session = requests_html.HTMLSession()
+    response = session.get(MINICONDA_REPO)
+    page: requests_html.HTML = response.html
+    table = page.find("table", first=True)
+    rows = table.find("tr")[1:]
+    for row in rows:
+        f, size, date, md5 = row.find("td")
+        fname = f.text
+        md5 = md5.text
+
+        if not fname.endswith(".sh"):
+            continue
+        stem = fname[:-3]
+
+        try:
+            s = MinicondaSpec.from_filestem(stem, md5)
+            if s.version.version_str != "latest":
+                yield s
+        except ValueError:
+            pass
+
+
+def key_fn(spec: MinicondaSpec):
+    return (
+        spec.version.version_str.info(),
+        spec.version.suffix.value,
+        spec.os.value,
+        spec.arch.value,
+    )
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
+    parser.add_argument("-d", "--dry_run", action="store_true")
```

Comment:
trivial - I would _feel_ that `--dry-run` is more intuitive than underscore name 🤔

Reply:
Yes, I dimly recall some time in the past argparse not being able to handle dashes in the middle of an option name but that doesn't seem to be the case now.